### DISCUSSION TITLE
Specify a bepoke db delegate with UpdateLockRowSemaphore

### DISCUSSION THREAD

**Discussion by neilatuniofyorkspace** [quartznet/quartznet#1201] [Q&A] [answered]
Is it possible to specify a bespoke db delegate with a

`UpdateLockRowSemaphore : DBSemaphore`

implementation ?

```
q.UsePersistentStore(store =>
{
    store.UseGenericDatabase("my-provider", db =>
    {
        db.UseDriverDelegate<MyDelegate>();
    });
});
```

**Comment by lahma** [comment, marked as answer]
Hmm... For some reason there's no code-configuration support for lock handlers (as far as I can tell). You should be able to configure it via normal property setting though, for example:

```csharp
s.SetProperty(
    StdSchedulerFactory.PropertyJobStoreLockHandlerType,
    typeof(UpdateLockRowSemaphore).AssemblyQualifiedNameWithoutVersion());
```
